### PR TITLE
operator: start the event queue in a dedicated go routine

### PR DESCRIPTION
**Pull request by aanm**
When splitting the event queue into a dedicated queue it was missed that
the queue should have been running in a separate go routine since the
queue handler will block the code execution.

Fixes: 5a4d26164998 ("operator: split event queue into two dedicated queues")
Signed-off-by: André Martins <andre@cilium.io>

### CODE REVIEW

File: operator/cilium_node.go [hunk 1]
```diff
@@ -223,6 +226,7 @@ func startSynchronizingCiliumNodes(ctx context.Context, nodeManager allocator.No
 		// then there isn't any event handler set for CiliumNodes events.
 		if withKVStore && kvStoreSyncHandler != nil {
 			<-connectedToKVStore
+			log.Info("Connected to the KVStore, syncing CiliumNodes to the KVStore")
 			for processNextWorkItem(kvStoreQueue, kvStoreSyncHandler) {
 			}
```
Comment (christarazi) [inline operator/cilium_node.go]: What about this one?

Comment (aanm) [reply operator/cilium_node.go]: That one is fine because it's part of the go routine initialized on line 208

**Review by julianwiedmann — APPROVED** [review]
Far from an expert on this part. But this clearly prevented us from publishing any CiliumNode updates to the KVStore. Thank you Andre!

File: operator/cilium_node.go (hunk 1 above)
Comment (christarazi) [reply operator/cilium_node.go]: I find that confusing. Could we add a comment to explain why? I'm not sure I understand why line 216 needs a goroutine and line 230 doesn't.

Comment (aanm) [reply operator/cilium_node.go]: @christarazi these statements are an infinite for loop, they will block the program execution. I've added some comments in the code.

**Review by christarazi — APPROVED** [review]

**Pull request merged**